src/rules.c

```diff
@@ -112,9 +112,9 @@ SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, StrongRef ref, QueryError *s
     rule->lang_default = DEFAULT_LANGUAGE;
   }
 
-  rule->prefixes = array_new(const char *, 1);
+  rule->prefixes = array_new(sds, 1);
```

Comment:
Shouldn't this be created with capacity for `nprefixes` elements? It seems a waste of time to create an array with capacity 1 and then `array_grow` n times.